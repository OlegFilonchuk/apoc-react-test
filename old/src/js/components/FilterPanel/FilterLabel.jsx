import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './FilterLabel.less';

export default function FilterLabel({ label, labelIcon, labelCount, className }) {
  const finalClassName = classNames('label-item', className);

  const labelItemIcon = labelIcon ? (
    <span className="label-item__icon">
      <i className={labelIcon} />
    </span>
  ) : null;

  return (
    <div className={finalClassName}>
      {labelItemIcon}
      <div className="label-item__name">{label}</div>
      {labelCount !== null && <div className="label-item__number">({labelCount})</div>}
    </div>
  );
}

FilterLabel.propTypes = {
  /**
   * Label for filter item
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),

  /**
   * Icon for filter item
   */
  labelIcon: PropTypes.string,

  /**
   * Label for filter item
   */
  labelCount: PropTypes.node,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

FilterLabel.defaultProps = {
  label: '',
  labelIcon: '',
  labelCount: null,
  className: ''
};
