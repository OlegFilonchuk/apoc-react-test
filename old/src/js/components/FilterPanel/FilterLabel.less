.label-item {
  display: flex;

  &__icon {
    margin-right: 5px;
  }

  &__name {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__number {
    margin-left: 5px;
  }
}
