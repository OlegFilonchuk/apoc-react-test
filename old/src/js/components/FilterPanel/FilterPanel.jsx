import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Panel from '../Panel/Panel';
import PanelHeader from '../Panel/PanelHeader';
import Checkbox from '../Checkbox/Checkbox';
import Multiselect from '../Multiselect/Multiselect';

import { PropTypeOption, updateOptions } from './FilterPanelCommon';

/* eslint-disable react/no-unused-prop-types */
export default class FilterPanel extends Component {
  componentWillMount() {
    updateOptions(this);
  }

  componentWillUpdate(nextProps) {
    updateOptions(this, nextProps);
  }

  render() {
    const { title, className } = this.props;

    return (
      <Panel modifierClass={className}>
        <PanelHeader>
          <h3>{title}</h3>
        </PanelHeader>
        {this.options}
      </Panel>
    );
  }
}

FilterPanel.propTypes = {
  /**
   * Options container this component will render - no need to change for now
   *  - defaults to `Multiselect`
   */
  OptionsContainer: PropTypes.func,

  /**
   * Single option component - no need to change for now
   *  - defaults to `Checkbox`
   */
  OptionComponent: PropTypes.func,

  /**
   * Title of the panel
   */
  title: PropTypes.node.isRequired,

  /**
   * Options for the filter
   */
  options: PropTypes.arrayOf(PropTypeOption).isRequired,

  /**
   * callback when filtering options have changed
   */
  onChange: PropTypes.func,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

FilterPanel.defaultProps = {
  OptionsContainer: Multiselect,
  OptionComponent: Checkbox,
  onChange: () => true,
  className: ''
};
