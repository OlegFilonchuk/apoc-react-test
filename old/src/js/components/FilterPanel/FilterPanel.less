.filter__label--truncate {
  .square-checkbox__label {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}
