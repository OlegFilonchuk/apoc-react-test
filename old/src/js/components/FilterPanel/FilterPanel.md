## Vertical example (default)

    const { makeSomeFilters } = require('./FilterPanelExamplesTools');

    <FilterPopover>
        {makeSomeFilters()}
    </FilterPopover>

## Horizontal example

    const { makeSomeFilters } = require('./FilterPanelExamplesTools');
    
    <FilterPopover horizontal>
        {makeSomeFilters()}
    </FilterPopover>
