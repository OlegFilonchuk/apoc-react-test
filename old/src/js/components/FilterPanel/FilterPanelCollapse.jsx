import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Collapse from '../Collapse/Collapse';
import Checkbox from '../Checkbox/Checkbox';
import Multiselect from '../Multiselect/Multiselect';

import { PropTypeOption, updateOptions, collapseIconStyles } from './FilterPanelCommon';

/* eslint-disable react/no-unused-prop-types */
export default class FilterPanelCollapse extends Component {
  componentWillMount() {
    updateOptions(this);
  }

  componentWillUpdate(nextProps) {
    updateOptions(this, nextProps);
  }

  render() {
    const { id, title, collapsed, onCollapseStateChange, className, displayCounter } = this.props;

    return (
      <Collapse
        className={className}
        title={title}
        isCollapsed={collapsed}
        onToggled={isCollapsed => onCollapseStateChange(id, isCollapsed)}
        iconStyles={collapseIconStyles}
        isPanelButton
        displayCounter={displayCounter}
      >
        {this.options}
      </Collapse>
    );
  }
}

FilterPanelCollapse.propTypes = {
  /**
   * Options container this component will render - no need to change for now
   *  - defaults to `Multiselect`
   */
  OptionsContainer: PropTypes.func,

  /**
   * Single option component - no need to change for now
   *  - defaults to `Checkbox`
   */
  OptionComponent: PropTypes.func,

  /**
   * id of the filter
   */
  id: PropTypes.string.isRequired,

  /**
   * Title of the panel
   */
  title: PropTypes.node.isRequired,

  /**
   * Is filter collapsed
   */
  collapsed: PropTypes.bool,

  /**
   * Options for the filter
   */
  options: PropTypes.arrayOf(PropTypeOption).isRequired,

  /**
   * callback when filtering options have changed
   */
  onChange: PropTypes.func,

  /**
   * Function that will run when filter collapse is changed
   */
  onCollapseStateChange: PropTypes.func,

  /**
   * Any custom class names
   */
  className: PropTypes.string,

  /**
   * If true, show counter for selected items
   */
  displayCounter: PropTypes.bool
};

FilterPanelCollapse.defaultProps = {
  OptionsContainer: Multiselect,
  OptionComponent: Checkbox,
  collapsed: false,
  onChange: () => true,
  onCollapseStateChange: () => true,
  className: '',
  displayCounter: false
};
