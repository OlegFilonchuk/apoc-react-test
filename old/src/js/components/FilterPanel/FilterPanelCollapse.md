## Vertical example (default)

    const { makeSomeFilters } = require('./FilterPanelExamplesTools');

    <FilterPopover>
        {makeSomeFilters(true)}
    </FilterPopover>

## Horizontal example

    const { makeSomeFilters } = require('./FilterPanelExamplesTools');
    
    <FilterPopover horizontal>
        {makeSomeFilters(true)}
    </FilterPopover>
