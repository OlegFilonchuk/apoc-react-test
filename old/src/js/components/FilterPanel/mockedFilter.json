{
  "title": "Some Filter Title",
  "id": "Some Filter Id",
  "data": [
    {
      "label": "app:app",
      "name": "Apps",
      "disabled" : false
    },
    {
      "label": "host",
      "name": "DUTs",
      "disabled" : false
    },
    {
      "label": "monitor",
      "name": "Monitors",
      "disabled" : true
    }
  ]
}
