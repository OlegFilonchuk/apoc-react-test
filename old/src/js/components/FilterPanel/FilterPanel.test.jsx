import React from 'react';
import { mount } from 'enzyme';

import mockedFilter from './mockedFilter.json';

import FilterPanel from './FilterPanel';

const mockTitle = mockedFilter.title;
const mockId = mockedFilter.id;

describe('<FilterPanel />', () => {
  const onChange = jest.fn();
  const onCollapseStateChange = jest.fn();

  const mountElement = options =>
    mount(
      <FilterPanel
        options={options}
        title={mockTitle}
        onChange={onChange}
        onCollapseStateChange={onCollapseStateChange}
        id={mockId}
      />
    );

  describe('default behavior', () => {
    const options = [{ id: 'test', label: 'test', name: 'test' }];

    let filter;

    beforeEach(() => {
      filter = mountElement(options);
    });

    it('should render FilterPanel component', () => {
      expect(filter).toMatchSnapshot();
    });

    it('should render FilterPanel component with whole panel not clickable/collapsable', () => {
      expect(filter.find('.panel__wrapper--collapsed').length).toEqual(0);

      filter.find('.sc-panel-heading').simulate('click');

      expect(filter.find('.panel__wrapper--collapsed').length).toEqual(0);
    });
  });

  describe('checked parameter', () => {
    const options = [
      {
        label: 'test',
        name: 'test',
        id: '1',
        checked: true
      },
      {
        label: 'test2',
        id: '2',
        name: 'test2'
      }
    ];

    const filter = mountElement(options);

    const testOptionCheckedStatus = (optionIndex, expectedIsChecked) => {
      const input = filter.find('.square-checkbox-square input').at(optionIndex);
      const isChecked = !!input.getElement().props.checked;

      expect(isChecked).toEqual(expectedIsChecked);
    };

    it('should be checked', () => {
      const optionIndex = 0;
      const expectedIsChecked = true;

      testOptionCheckedStatus(optionIndex, expectedIsChecked);
    });

    it('should not be checked', () => {
      const optionIndex = 1;
      const expectedIsChecked = false;

      testOptionCheckedStatus(optionIndex, expectedIsChecked);
    });
  });
});
