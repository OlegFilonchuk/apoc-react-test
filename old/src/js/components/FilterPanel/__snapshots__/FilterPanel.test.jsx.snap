// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterPanel /> default behavior should render FilterPanel component 1`] = `
<FilterPanel
  OptionComponent={[Function]}
  OptionsContainer={[Function]}
  className=""
  id="Some Filter Id"
  onChange={[Function]}
  onCollapseStateChange={[Function]}
  options={
    Array [
      Object {
        "id": "test",
        "label": "test",
        "name": "test",
      },
    ]
  }
  title="Some Filter Title"
>
  <Panel
    modifierClass=""
  >
    <div
      className="sc-panel "
    >
      <PanelHeader
        className=""
        key=".0"
      >
        <div
          className="sc-panel-heading"
        >
          <div
            className="sc-panel-title"
          >
            <h3
              key=".0"
            >
              Some Filter Title
            </h3>
          </div>
        </div>
      </PanelHeader>
      <div
        className="sc-panel-body"
      >
        <Multiselect
          className=""
          fullWidth={true}
          key=".1"
          labelPosition="left"
          noVerticalLines={false}
          onChange={[Function]}
        >
          <ul
            className="multiselect"
          >
            <li
              className="multiselect__row"
              key=".$test"
            >
              <div
                className="li-content"
              >
                <Checkbox
                  checked={false}
                  className="filter__label--truncate"
                  data-test-element="testCheckbox"
                  disabled={false}
                  fullWidth={true}
                  isOnlyCheckboxClickable={false}
                  key="test"
                  label={
                    <FilterLabel
                      className=""
                      label="test"
                      labelCount={null}
                      labelIcon=""
                    />
                  }
                  labelPosition="left"
                  name="test"
                  onChange={[Function]}
                  showHint={true}
                  tooltip=""
                >
                  <div
                    className="square-checkbox filter__label--truncate square-checkbox--clickable square-checkbox-full-width"
                  >
                    <label
                      className="square-checkbox__label square-checkbox__label--clickable"
                      htmlFor="test"
                      title=""
                    >
                      <FilterLabel
                        className=""
                        label="test"
                        labelCount={null}
                        labelIcon=""
                      >
                        <div
                          className="label-item"
                        >
                          <div
                            className="label-item__name"
                          >
                            test
                          </div>
                        </div>
                      </FilterLabel>
                    </label>
                    <label
                      className="square-checkbox-square"
                    >
                      <input
                        checked={false}
                        data-test-element="testCheckbox"
                        disabled={false}
                        id="test"
                        name="test"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span />
                    </label>
                  </div>
                </Checkbox>
              </div>
            </li>
          </ul>
        </Multiselect>
      </div>
    </div>
  </Panel>
</FilterPanel>
`;
